cli/cmd/compose/ps.go

```diff
@@ -59,17 +61,26 @@ func runPs(ctx context.Context, opts composeOptions) error {
 		return err
 	}
 
-	err = printSection(os.Stdout, func(w io.Writer) {
-		for _, service := range serviceList {
-			fmt.Fprintf(w, "%s\t%s\t%d/%d\t%s\n", service.ID, service.Name, service.Replicas, service.Desired, strings.Join(service.Ports, ", "))
-		}
-	}, "ID", "NAME", "REPLICAS", "PORTS")
-	return err
+	return printPsFormatted(opts.Format, os.Stdout, serviceList)
 }
 
-func printSection(out io.Writer, printer func(io.Writer), headers ...string) error {
-	w := tabwriter.NewWriter(out, 20, 1, 3, ' ', 0)
-	fmt.Fprintln(w, strings.Join(headers, "\t"))
-	printer(w)
-	return w.Flush()
+func printPsFormatted(format string, out io.Writer, serviceList []compose.ServiceStatus) error {
+	var err error
+	switch strings.ToLower(format) {
+	case formatter.PRETTY, "":
+		err = formatter.PrintPrettySection(out, func(w io.Writer) {
+			for _, service := range serviceList {
+				fmt.Fprintf(w, "%s\t%s\t%d/%d\t%s\n", service.ID, service.Name, service.Replicas, service.Desired, strings.Join(service.Ports, ", "))
+			}
+		}, "ID", "NAME", "REPLICAS", "PORTS")
+	case formatter.JSON:
+		outJSON, err := formatter.ToStandardJSON(serviceList)
```

Comment:
Same question here, although it does not expose too many additional things. At least we should set json attribut names in the go struct, current field case is mainly driven by public/private fields and go linter, that should not drive the json output